src/ethereum_test_tools/vm/opcode.py

```diff
@@ -188,6 +190,54 @@ def __str__(self) -> str:
         """
         return self._name_
 
+    def __eq__(self, other):
+        """
+        Required to differentiate between SELFDESTRUCT and SENDALL type of cases
+        And to register the Macro opcodes which are defined from same bytes
+        """
+        if isinstance(other, Opcode):
+            return self._name_ == other._name_
+        return NotImplemented
+
+
+class Macro(Opcode):
```

Comment:
I still don't quite get the point of `Macro` inheriting `Opcode`.

In my opinion, I feel like it should be it's own thing separate from `Opcode`:
```python
class Macro(bytes):
    ...

class Macros(Macro, Enum):
   ...
```

I think the only advantage of having macros in the same `Enum` is that you don't have to import two different things when writing the test, but is that enough reason to have them joined together?

I'm open to discussion, is just that I wanted to see the arguments.

Reply:
Yes. I don't want to write unions everywhere.

It is easy to import, define types, and write compare statements if it all lives just in one enum.

Reply:
I like this addition! 

But I also don't think `OOG` should be in `Opcode`, nor should `Macro` inherit from `Opcode`. Rather `Macros` should be a separate enum class. 

I don't like how this reads as though `OOG` is actually an opcode; I'd rather be explicit:
https://github.com/ethereum/execution-spec-tests/blob/d27fe5065db560648ae7bf8edad2d182b03a8935/tests/cancun/eip1153_tstore/test_tload_reentrancy.py#L30-L31

I don't think it's worth sacrificing being explicit for the sake of saving an import. With regards to typehints, we can define a new type (name suggestions welcome! :) )
```python
OpcodeOrMarco = Opcode | Macro
```